addons/website/models/website.py

```diff
@@ -612,26 +612,51 @@ def _render_snippet(key):
             snippets_cache[key] = data
             return data
 
+        text_generation_target_lang = self.get_current_website().default_lang_id.code
+        nb_terms_translated, nb_terms_not_translated = 0, 0
         generated_content = {}
         for page_code in requested_pages - {'privacy_policy'}:
             snippet_list = snippet_lists.get(page_code, [])
             for snippet in snippet_list:
                 render, placeholders = _render_snippet(f'website.{snippet}')
+                if text_generation_target_lang != 'en_US':
+                    # Check if terms are translated.
+                    translation_dictionary = self.env['website.page']._fields['arch_db'].get_translation_dictionary(
+                        IrQweb._render(f'website.{snippet}', cta_data, lang="en_US").__str__(),
+                        {text_generation_target_lang: render.__str__()},
+                    )
+                    for from_lang_term, to_lang_terms in translation_dictionary.items():
+                        if from_lang_term == to_lang_terms[text_generation_target_lang]:
+                            nb_terms_not_translated += 1
+                        else:
+                            nb_terms_translated += 1
```

Comment:
> I would have to have another variable like nb_total_translated_terms

That's what I meant of course, yes. Feels weird to me to count items that are X and the items that are not X instead of just counting the ones that are X while you already have in your hands the total amout of items per snippets.

\+ the use of a list comprehension instead of the loop + if + else + `+= 1` which is way more python-like to me (but again, not an expert).

But as you wish, you decide :)